### PR TITLE
Api versioning

### PR DESCRIPTION
**Pull request by romppane**
Route versioning added to the start of the url and temporary usernames added to the system with migrations!

### CODE REVIEW

File: server/entities/comment.ts [hunk 1]
```diff
@@ -42,16 +42,30 @@ export class Comment {
   @MaxLength(500)
   private message: string;
 
+  @Column({ length: 18 })
+  @Expose()
+  @IsAlphanumeric()
```
Comment (henrikra) [inline server/entities/comment.ts]: This prevents me from getting Henrik_R username :) Isn't string check enough?

File: server/entities/post.ts [hunk 2]
```diff
@@ -53,6 +53,12 @@ export class Post {
   @Length(1, 1000)
   private message: string;
 
+  @Column({ length: 18 })
+  @Expose()
+  @IsAlphanumeric()
```
Comment (henrikra) [inline server/entities/post.ts]: Same thing here

**Comment by henrikra** [comment]
Just update the docs on README about the endpoint changes and comment weird transition keys like ownerID :D that is only requirement

Also like we discussed that you decide on what level of validation you want on username and also where you want to put api version number in the url 👍

File: server/types.ts [hunk 3]
```diff
@@ -23,11 +24,7 @@ export type Dependencies = {
   commentRepository: Repository<Comment>;
   commentService: CommentService;
   errorHandler: ErrorRequestHandler;
-  validatePost: RequestHandler;
-  validatePostPUT: RequestHandler;
-  validateParams: RequestHandler;
-  validateComment: RequestHandler;
```
Comment (henrikra) [inline server/types.ts]: Remember to update di file too since you removed these :D

**Comment by romppane** [comment]
I've decided to stick with the original format of versioning! 
While making small changes this way may seem unnecessary drastic, I figured breaking changes most of the time probably are quite the opposite of small! This way it will be easier and more logical for me when maintaining both version if compared to clustering all the routes under one controller and letting it grow stupidly large!

**Pull request merged**